web/src/message_list_view.js

```diff
@@ -84,6 +86,15 @@ export function message_hover($message_row) {
         data_template_id = "move-message-tooltip-template";
     }
     $edit_content.attr("data-tooltip-template-id", data_template_id);
+
+    if (message_edit.is_content_editable(message)) {
+        update_edit_button_state = setInterval(() => {
+            if (!message_edit.is_content_editable(message)) {
+                clearInterval(update_edit_button_state);
+                message_hover($message_row);
+            }
+        }, 1000);
```

Comment:
This is super expensive -- an idle Zulip app is going to be checking this every second that you left the mouse open, which can preventing a laptop from fully spinning down its CPU.

What we should instead do is just calculate the time remaining and do a `setTimeout` -- storing the timer object in a variable so we can cancel it in the `unhover` function.

Reply:
Or maybe better would be to decide that this commit's corner case isn't a bug worth fixing directly; it's the kind of bug that a nicer web framework will fix automatically and doesn't really matter in that it only comes up if you leave your mouse over a message at the moment it's edit deadline runs out -- it's fixed if you move the mouse off and on again.